internal/controller/postgrescluster/postgres.go

```diff
@@ -621,6 +622,26 @@ func (r *Reconciler) reconcilePostgresDataVolume(
 
 	pvc.Spec = instanceSpec.DataVolumeClaimSpec
 
+	// If a source cluster was provided and VolumeSnapshots are turned on in the source cluster
+	// and there is a VolumeSnapshot available for the source cluster that is ReadyToUse, use it
+	// as the source for the PVC. If there is an error when retrieving VolumeSnapshots, log it,
+	// but continue creating PVC in the usual fashion.
+	if sourceCluster != nil && sourceCluster.Spec.Backups.Snapshots != nil && feature.Enabled(ctx, feature.VolumeSnapshots) {
+		snapshots, err := r.getSnapshotsForCluster(ctx, sourceCluster)
+		if err == nil {
+			snapshot := getLatestReadySnapshot(snapshots)
+			if snapshot != nil {
+				pvc.Spec.DataSource = &corev1.TypedLocalObjectReference{
+					APIGroup: initialize.String("snapshot.storage.k8s.io"),
+					Kind:     "VolumeSnapshot",
+					Name:     snapshot.Name,
+				}
+			}
+		} else {
+			log.Error(err, fmt.Sprintf("Could not get snapshots for %v, proceeding with typical restore process.", sourceCluster.Name))
```

Comment:
🔧 If we want people to see this, it should be a warning event on `cluster` instead.